src/attributedlabel/unittests/NIAttributedLabelTests.m

```diff
@@ -16,18 +16,97 @@
 
 // See: http://bit.ly/hS5nNh for unit test macros.
 
+#import <Foundation/Foundation.h>
 #import <XCTest/XCTest.h>
 
 #import "NimbusAttributedLabel.h"
+#import "NIAttributedLabel+Testing.h"
+
+@interface NIAttributedLabelTestMonitor : NSObject <NIAttributedLabelDelegate>
+
+@property(nonatomic, assign) BOOL doNotRespondToLongPressCallback;
+
+@property(nonatomic, readonly) int longPressCallbackCount;
+@property(nonatomic, readonly) int actionSheetCallbackCount;
+
+@end
+
+@implementation NIAttributedLabelTestMonitor
+
+- (BOOL)respondsToSelector:(SEL)selector {
+  if (selector == @selector(attributedLabel:didLongPressTextCheckingResult:atPoint:)) {
+    return !_doNotRespondToLongPressCallback;
+  }
+
+  return [super respondsToSelector:selector];
+}
+
+- (void)attributedLabel:(NIAttributedLabel *)attributedLabel didLongPressTextCheckingResult:(NSTextCheckingResult *)result atPoint:(CGPoint)point {
+  _longPressCallbackCount++;
+}
+
+- (BOOL)attributedLabel:(NIAttributedLabel *)attributedLabel shouldPresentActionSheet:(UIActionSheet *)actionSheet withTextCheckingResult:(NSTextCheckingResult *)result atPoint:(CGPoint)point {
+  _actionSheetCallbackCount++;
+  return NO;
+}
+
+@end
 
 @interface NIAttributedLabelTests : XCTestCase
 @end
 
 
 @implementation NIAttributedLabelTests
 
+- (void)testLongPressCallbackOverridesActionSheetCallbackWhenTouchingLink {
+  // Create an attributed string with a link.
+  NSString *string = @"NimbusKit";
+  NSURL *URL = [NSURL URLWithString:@"http://nimbuskit.info/"];
+  NSTextCheckingResult *textCheckingResult = [NSTextCheckingResult linkCheckingResultWithRange:NSMakeRange(0, string.length) URL:URL];
+  NSDictionary<NSAttributedStringKey,id> *attributes = @{ NIAttributedLabelLinkAttributeName : textCheckingResult };
+  NSAttributedString *attributedText = [[NSAttributedString alloc] initWithString:string attributes:attributes];
```

Comment:
Should this common attributed string creation be pulled into a setUp and nil'ed out in tearDown?

Reply:
Maybe also with the Label below. It could all be pulled into common creation.

Reply:
I don't think it makes sense to pull this initialization into -setUp and -tearDown; the individual tests have everything they need in their method scope.

I considered pulling common initialization into a helper method but I think the tests read more clearly with the initialization inlined as it is now. I think that a helper method that returned a label and a text checking result could eventually be beneficial but I think I would prefer to add that later. Also it may be relevant to point out that the primary goal here was to explicitly initialize things to exercise in tests and that it is an accident (perhaps an accident of convenience) that the logic is the same.

Reply:
Fair enough.